CMakeLists.txt

```diff
@@ -89,12 +89,11 @@ if (BUILD_PIPS_IPM OR BUILD_PIPS_NLP)
   find_package(OpenMP)
 endif()
 
-# include different "whole archive" linking options depending on compiler
-if (CMAKE_CXX_COMPILER_ID MATCHES "^(Apple)?Clang$")
-  set(WHOLE_ARCHIVE "-Wl,-all_load")
-  set(NO_WHOLE_ARCHIVE "-Wl,-noall_load")
-  message(STATUS "SETTING HAVE_CLANG")
-  set(HAVE_CLANG TRUE)
+# include different "whole archive" linking options depending on system
+if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
```

Comment:
How is different from `if(APPLE)`? Inside this `if` condition, there is `if(APPLE)` too.